### PR TITLE
  Add --requires flag to podman run/create 

### LINKED ISSUE #9250: Expose `--dependency` for `podman run`
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**

There's already a field in the container metadata (`podman inspect xxx`) called `Dependencies`, which could perfectly solve the problem of container dependency (e.g. apps should be run after the database). But there's no such way to add dependency when create the container. The only place it's used now is for the pod `pause` container.

**Steps to reproduce the issue:**

1. `podman run --dep postgres alpine:latest`

**Describe the results you received:**

No such a parameter

**Describe the results you expected:**

There is such a parameter

### PR DESCRIPTION
**Pull request by mheon**
Podman has, for a long time, had an internal concept of
dependency management, used mainly to ensure that pod infra
containers are started before any other container in the pod. We
also have the ability to recursively start these dependencies,
which we use to ensure that `podman start` on a container in a
pod will not fail because the infra container is stopped. We have
not, however, exposed these via the command line until now.

Add a `--requires` flag to `podman run` and `podman create` to
allow users to manually specify dependency containers. These
containers must be running before the container will start. Also,
make recursive starting with `podman start` default so we can
start these containers and their dependencies easily.

Fixes #9250

Signed-off-by: Matthew Heon <matthew.heon@pm.me>

### CODE REVIEW

**Comment by Luap99** [comment]
I just tried this and when I try to remove a container that another container depend on, I get this error: `Error: container 94094b2994a235cac4a4bb130ea4448bdce45a6043e84ebb53719d6b283726ec has dependent containers which must be removed before it: eec4909c35b11d63ec1c643cb4b293241aedd5e270af4c385243c275127ca56e: container already exists` The `container already exists` part makes no sense. I think this should be changed or removed.

**Comment by mheon** [comment]
Sure, a new error here sounds reasonable. I also need to get Swagger in order.

**Comment by mheon** [comment]
Tests have gone green. @containers/podman-maintainers PTAL

File: test/e2e/run_test.go
Comment (TomSweeneyRedHat) [inline test/e2e/run_test.go:1575]: maybe not in this test, but it would be good to have 2 or 3 dependency containers.

File: docs/source/markdown/podman-create.1.md
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-create.1.md:253]: How do they specify?  name, ctrid, other?  How are multiples specified?  One per `--dep` option or by comma separated list?  Examples would be very nice.  Same comment through out.

File: docs/source/markdown/podman-create.1.md
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-create.1.md:255]: need to add podman-start(1) to the "SEE ALSO" list.

Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-create.1.md:255]: Very soft suggestion for consideration.  I'm not married to it.
```suggestion
The `podman start` command can start these dependencies first using the `--recursive` flag.
```

File: docs/source/markdown/podman-run.1.md
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-run.1.md:272]: I find this sentence to be confusing.  You're saying run automatically starts the dependencies, but the start command must be called with the recursive option first?  Won't that start command start the containers when it's called?

Would it make sense to add a `--recursive` option to run and just forgo the start command if you're doing a run (or perhaps call it under the covers?)

File: docs/source/markdown/podman-create.1.md
Comment (TomSweeneyRedHat) [reply docs/source/markdown/podman-create.1.md:255]: same comment as applicable to the other commands.

File: docs/source/markdown/podman-create.1.md
Comment (TomSweeneyRedHat) [reply docs/source/markdown/podman-create.1.md:253]: same comment for the other commands.

**Comment by mheon** [comment]
Updated, comments should be addressed

File: docs/source/markdown/podman-run.1.md
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-run.1.md:1638]: :+1:

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Review by vrothberg — COMMENTED** [review]
Code LGTM but I'd love to have a more complex testing scenario with a longer and branched graph.

One idea I want to throw out is to make the two new flags more symmetrical. `--dep` and `--recursive` do not immediately suggest a relation. Something I'd find more approachable would be `podman create --name newContainer --dependency anotherContainer` and `podman start --dependencies newContainer` (which would start anotherContainer). So it would be `--dependency` instead of `--dep` and `--dependencies` instead of `--recursive`.

Not at all a blocker but a preference.

**Comment by rhatdan** [comment]
Well we should settle on the flag name before merging.  Could we discuss this at standup today.

**Comment by mheon** [comment]
Flag name changed to `--requires` and `--recursive` argument to `podman start` dropped and made default, per discussion. Should be ready now @containers/podman-maintainers

File: docs/source/markdown/podman-create.1.md
Comment (rhatdan) [inline docs/source/markdown/podman-create.1.md:1266]: The --recursive option no longer exists.

File: docs/source/markdown/podman-create.1.md
Comment (rhatdan) [inline docs/source/markdown/podman-create.1.md:1271]: The --recursive option no longer exists.

File: docs/source/markdown/podman-create.1.md
Comment (rhatdan) [inline docs/source/markdown/podman-create.1.md:1280]: The --recursive option no longer exists.

File: docs/source/markdown/podman-create.1.md
Comment (rhatdan) [inline docs/source/markdown/podman-create.1.md:1279]: --dep->--requires

File: docs/source/markdown/podman-run.1.md
Comment (rhatdan) [inline docs/source/markdown/podman-run.1.md:1626]: --requires

File: docs/source/markdown/podman-run.1.md
Comment (rhatdan) [inline docs/source/markdown/podman-run.1.md:1634]: --requires

File: docs/source/markdown/podman-create.1.md
Comment (rhatdan) [inline docs/source/markdown/podman-create.1.md:1270]: --requires.

**Review by rhatdan — APPROVED** [review]

**Comment by mheon** [comment]
Manpages touched up

**Review by rhatdan — APPROVED** [review]
LGTM

**Comment by mheon** [comment]
Ready to merge

**Review by ashley-cui — COMMENTED** [review]
/lgtm

**Pull request merged**
